### PR TITLE
RE-1329 Compress Logs

### PR DESCRIPTION
**Pull request by hughsaunders**
This commit ensures console logs are zipped when a job terminates.
This will affect all future jobs, current jobs are currentbly being
compressed by a bash loop. This relies on a plugin which has already
been installed.

Issue: [RE-1329](https://rpc-openstack.atlassian.net/browse/RE-1329)

### CODE REVIEW

**Comment by hughsaunders** [comment]
Created job with DSL
```groovy

properties([compressBuildLog()])
print("test compression plugin")
```
 https://rpc.jenkins.cit.rackspace.net/job/scratchpipeline/830/console
Also verified via cli that the console is compressed.

**Comment by hughsaunders** [comment]
Marked as WIP as this needs a new periodic to compress the blue ocean logs.

**Comment by hughsaunders** [comment]
Introduced second commit which adds a section to periodic cleanup that compresses stage logs.
Ready for review.

File: rpc_jobs/periodic_cleanup.yml [hunk 1]
```diff
@@ -117,4 +117,23 @@
...
+            // Note that already compressed logs will be skipped as they
+            // get renamed .gz .
+            // This only affects the stage logs used for blue ocean,
+            // console logs are compressed using a plugin activated via
+            // common.globalWraps.
+            // Logs newer than 1 day are skipped to prevent compressing
+            // the logs of jobs that are currently running.
+            sh """#!/bin/bash
+              cd /var/lib/jenkins/jobs
+              find . -name \\*.log -mtime +1 \\
```
Comment (git-harry) [inline rpc_jobs/periodic_cleanup.yml]: It looks like there are other things, e.g. workspaces, in the job directories in addition to log files. While unlikely, it seem like there is the potential for this to modify a live build. What are your thoughts on something a bit more targetted, e.g. `find . -regex '.+/builds/[0-9]+/[0-9]+\.log' -mtime +1`

**Comment by hughsaunders** [comment]
Updated to include the tighter regex suggested by @git-harry.

**Pull request merged**